src/integration-test/java/eu/xenit/gradle/testrunner/AbstractIntegrationTest.java

```diff
@@ -24,7 +26,12 @@ protected void testProjectFolderThatShouldFail(Path projectFolder, String task)
     protected void testProjectFolder(Path projectFolder, String task)throws IOException {
         testProjectFolder(projectFolder,task,false);
     }
-    protected void testProjectFolder(Path projectFolder, String task, boolean expectsException) throws IOException {
+
+    protected  void testProjectFolder(Path projectFolder, String task, boolean expectsException) throws IOException {
+        testProjectFolder(projectFolder, task, expectsException, null);
+    }
+
+    protected void testProjectFolder(Path projectFolder, String task, boolean expectsException, String message) throws IOException {
```

Comment:
Wouldn't it be better to separate this functionality into an other method, which asserts a failure?
Or even pass a `Consumer<BuildResult>` as a parameter so you can do your own checks when it is needed.